### PR TITLE
EN: CLI: Update

### PR DESCRIPTION
**Pull request by mlep**
For review.

What about BibTeX vs BibLaTex?  (not a word about BibLaTeX at the moment in this help file...)

### CODE REVIEW

File: en/CommandLine.md [hunk 1]
```diff
@@ -1,77 +1,174 @@
...
-## Help: -h
+java -jar JabRef.jar [OPTIONS] [BIBTEX_FILE]
 
-This option causes JabRef to display a summary of the command line options, and quit immediately.
+*Note:* you may have to replace `JabRef.jar` by the actual name of the `.jar` file. 
 
-## No-GUI mode: -n
+You can always specify one or more BibTeX files to load by simply listing their filenames.
+Take care to specify all options before your list of file names.
+You must make sure that the first file name is not misunderstood as being an argument for an option; this simply means that if a boolean option like `-n` or `-l` immediately precedes a file name, add the word `true` as an argument.
```
Comment (Siedlerchr) [inline en/CommandLine.md]: Maybe "Ensure that the first file name....
instead of must make sure

File: en/CommandLine.md [hunk 2]
```diff
@@ -1,77 +1,174 @@
...
+## Options
+
+`[-a <FILE>] [-asfl] [-b] [-d <FILE>] [--debug] [-f <FILE>] [-g]
+       [-h] [-i <FILE>] [--importToOpen <FILE>] [-m <FILE>] [-n] [-o
+       <FILE>] [-p <FILE>] [-v] [-x <FILE>]`
+
+### Help: `-h`
+(or `--help`)
+
+Displays a summary of the command line options (including the list of available import and export formats), and quit immediately.
```
Comment (Siedlerchr) [inline en/CommandLine.md]: I would remove the last sentence ("and quot...)
This typical behavior of a command line, so so need to describe it

File: en/CommandLine.md [hunk 3]
```diff
@@ -1,77 +1,174 @@
...
+
+
+### No-GUI mode: `-n`
+(or `--nogui`)
+
+Suppresses the JabRef window (i.e. no GUI - Graphic User Interface - is displayed).
+
+It causes the program to quit immediately once the command line options have been processed.
```
Comment (Siedlerchr) [inline en/CommandLine.md]: I like "exit" better than "quit" in the context of a program

File: en/CommandLine.md [hunk 4]
```diff
@@ -1,77 +1,174 @@
...
-## Import file: -i filename\[,format\]
 
-This option causes JabRef to import or load the specified file. If only a filename is specified, or the filename is followed by a comma and a \* character, JabRef will attempt to detect the file format automatically. This should work for all BibTeX files and files in any import format supported by JabRef. If the filename is followed by a comma and the name of an import format, the given import filter will be used. Use the `-h` option to get a list of availiable import formats.
+### Import file: `-i filename[,import format]`
+(or `--import filename[,import format]` or `--importToOpen filename[,import format]`)
+
+Import or load the file `filename`.
+
+If only the filename is specified (or if the filename is followed by a comma and a `*` character), JabRef will attempt to detect the file format automatically.
+This should work for all BibTeX files, and also for files in any import format supported by JabRef.
```
Comment (Siedlerchr) [inline en/CommandLine.md]: "Should work" -> This works.... and also for all files in a supported input format ....

File: en/CommandLine.md [hunk 5]
```diff
@@ -1,77 +1,174 @@
...
+If the filename is followed by a comma and the name of an import format, the given import filter will be used.
+
+Use the `-h` option to get the list of available import formats.
+
+If an export option is also specified, the import will always be processed first, and the imported or loaded file will be used by the export filter.
+If the GUI is not suppressed (using the `-n` option), any imported or loaded file will show up in the main window.
+
+If `--importToOpen` is used, the content of the file will be imported to the opened tab.
```
Comment (Siedlerchr) [inline en/CommandLine.md]: imported into

File: en/CommandLine.md [hunk 6]
```diff
@@ -1,77 +1,174 @@
...
+`(author=bock or title|keywords="computer methods")and not(author=sager)`
+
+
+### Fetch entries from Web: `-f=FetcherName:QueryString`
+(or `--fetch=FetcherName:QueryString`)
+
+Query a Web fetcher and import the entries.
+
+Pass both the name of a fetcher and your search term or paper id (e.g. `--fetch=Medline:cancer`), and the given fetcher will be run.
+Some fetchers will still display GUI if they need feedback from you.
```
Comment (Siedlerchr) [inline en/CommandLine.md]: a GUI window

File: en/CommandLine.md [hunk 7]
```diff
@@ -1,77 +1,174 @@
...
+The fetchers listed in the Web search panel can be run from the command line.
+To get the list of available fetchers, run `--fetch` without parameters.
+
+
+### Subdatabase from .aux file: `-a infile[.aux],outfile[.bib] base-BibTeX-file`
+(or `--aux infile[.aux],outfile[.bib] base-BibTeX-file`)
+
+Extract a subdatabase from a .aux file:
+
+When you compile a LaTeX document (e.g. `infile.tex`), a .aux file is created (`infile.aux`).
```
Comment (Siedlerchr) [inline en/CommandLine.md]: an .aux file

**Comment by Siedlerchr** [comment]
Looks good.  Regarding Biblatex, we should just mention it somewhere in the index/start page of the help.
Or maybe a point in the FAQ : Does JabRef support BibLateX ? 
(https://www.ctan.org/pkg/biblatex)

**Comment by Siedlerchr** [comment]
Thanks for your good work! :+1:  Just some minor rewording and we can merge it in. 

Regarding Biblatex, we should just  add a point in the FAQ:  Does JabRef support BibLateX ? 
and then we could link the official biblatex manual

File: en/CommandLine.md (hunk 1 above)
Comment (mlep) [inline en/CommandLine.md]: Done.

File: en/CommandLine.md (hunk 3 above)
Comment (mlep) [inline en/CommandLine.md]: Done.

File: en/CommandLine.md (hunk 4 above)
Comment (mlep) [inline en/CommandLine.md]: Done.

File: en/CommandLine.md (hunk 6 above)
Comment (mlep) [inline en/CommandLine.md]: Done.

File: en/CommandLine.md (hunk 7 above)
Comment (mlep) [inline en/CommandLine.md]: Done.

Comment (oscargus) [inline en/CommandLine.md]: Or "a dot-aux" file. Not really sure.

File: en/CommandLine.md [hunk 8]
```diff
@@ -1,77 +1,174 @@
...
+`java -jar JabRef.jar -o filetoexport.xml,docbook -n true original.bib`
 
+will correctly load the file `original.bib`, export it in docbook format to `filetoexport.xml`, and suppress the GUI.
+The word *true* prevents the file name from being interpreted as an argument to the `-n` option.
+
+
+## Options
+
+`[-a <FILE>] [-asfl] [-b] [-d <FILE>] [--debug] [-f <FILE>] [-g]
+       [-h] [-i <FILE>] [--importToOpen <FILE>] [-m <FILE>] [-n] [-o
```
Comment (oscargus) [inline en/CommandLine.md]: Based on this there is no argument to the `-n` option. I have no idea how it actually works though.

File: en/CommandLine.md [hunk 9]
```diff
@@ -1,77 +1,174 @@
...
+### Version: `-v`
+(or `--version`)
 
-## Export only used items: -a filename\[.aux\],newBibFile\[.bib\]
+Display the version number of JabRef.
 
-Sometimes it is helpful, to have a bibtex file that contains only the used bibtex entries. A list of these used entries is located in an aux file. Jabref can parse this file to generate a new bibtex file, which contains only the known and used entries. That will mean, if an entry is not defined in the standard bibtex file, it cannot be located in the new file.
 
-## Fetch from Web: --fetch==name of fetcher:query string
+### Debug mode: `--debub`
```
Comment (oscargus) [inline en/CommandLine.md]: --debug?

**Comment by oscargus** [comment]
Just some minor comments. It is great that you are doing this @mlep !

**Comment by mlep** [comment]
@Siedlerchr : Thank you for reviewing this help file.

About BibLaTeX, my issue is more about: 
- What happens if you feed JabRef's CLI with a BibLaTeX file?
- Does everything written about BibTeX stands?
- Anc, currently, in the CLI help (`jabref --help`), there is no import or export format for BibLaTeX...

File: en/CommandLine.md (hunk 8 above)
Comment (mlep) [inline en/CommandLine.md]: `-n` works well if there is no BibTeX file mentioned on the command line.
In the meantime, "The word *true* prevents the file name from being interpreted as an argument to the `-n` option" is kind of weird. I do not remember seeing something like this in any command I have ever used. So, a JabRef *very-specific* feature? :smile:

**Comment by mlep** [comment]
@oscargus Thanks. You are welcome.

File: en/CommandLine.md (hunk 9 above)
Comment (mlep) [inline en/CommandLine.md]: Sure!
Corrected.

File: en/CommandLine.md (hunk 7 above)
Comment (mlep) [inline en/CommandLine.md]: Not really sure either...

Comment (Siedlerchr) [inline en/CommandLine.md]: Me neither ;) No idea.

**Comment by Siedlerchr** [comment]
@mlep  Regarding BibTeX or BibLaTex for JabRef: It is the same bib file format (okay, JabRef adds the meta comment, but that is just for setting the  mode in JabRef).  It is not an extra file format like xml or so.
In short: BibLatTeX has just some more entry types (e.g. Patent, Online) and different fields than the old bibtex format.  And you can mix both fields in bib files (backward compatible)

For JabRef it is really just those field/entry type differences. Only for Latex users the other things play crucial role, regarding layout of citation styles, macros etc, .e.g in how to display the citation/reference list.

File: en/CommandLine.md [hunk 10]
```diff
@@ -1,77 +1,174 @@
...
-Examples
 
--   `Year=2005`
--   `title|keywords=Optimization`
--   `(author=bock or title|keywords="computer methods")and not(author=sager)`
--   `Year=1989-2005`
+### Version: `-v`
+(or `--version`)
 
-## Export only used items: -a filename\[.aux\],newBibFile\[.bib\]
```
Comment (koppor) [inline en/CommandLine.md]: Doesn't that work anymore?

I have to take time to really review this PR. Hope, I find time next week.

**Pull request merged**